packages/web-client/app/models/animated-workflow.ts

```diff
@@ -17,7 +17,15 @@ let waiter = buildWaiter('thread-animation');
 let token: any = null;
 let waiting: RSVP.Deferred<void> | null = null;
 
-let interval = config.environment === 'test' ? 100 : 1000;
+let interval = 1000;
+if (config.environment === 'test') {
+  interval = 0;
+} else if (
+  config.environment === 'development' &&
+  config.threadAnimationInterval
+) {
+  interval = config.threadAnimationInterval;
+}
```

Comment:
I wonder if it might be preferable to move as much of this as possible into `config/environment.js`? Since it already has blocks that override defaults for `development` and `test`, it might be nice to concentrate things there.

Also, a possible development productivity improvement would be to allow a query parameter override so one can change `threadAnimationInterval` without needing to stop the server and update `THREAD_ANIMATION_INTERVAL`. I’ve done this a few times, like:
* `?faker-seed=1234` to [change the randomisation seed at runtime](https://github.com/hashicorp/nomad/blob/0af476263b5c6f12a43a069070400e1762ad17c2/ui/mirage/faker.js#L4-L20) instead of build time
* `?mirage-logging=true` to [enable Mirage logging](https://github.com/hashicorp/nomad/blob/a9ec864a1ab47030761788ac11bd8d85ebd43145/ui/mirage/config.js#L27) without needing to restart

I haven’t been working with the animations enough yet to know whether this is something that would be tweaked enough to make a query parameter override useful, so this could be something to revisit later with more experience. Maybe it’s overkill, I just have a distaste for stopping and starting the server to adjust things haha. If this does seem worthy, despite what I said above I believe the override would have to happen here, at consumption, vs in `config/environment`, which doesn’t live-update.

Reply:
Very interesting, hadn't considered query params as a way to control this. Thanks for the suggestions, might need some time to digest and act on them!